### PR TITLE
Regex change for #7194

### PR DESCRIPTION
**Pull request by rgstephens**
**Proposed changes**:
- For issue #7194, change the regex to exclude processing of payloads with entities.
- fix https://github.com/RasaHQ/rasa/issues/7194

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/core/nlg/interpolator.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ def interpolate_text(template: Text, values: Dict[Text, Text]) -> Text:
     """
 
     try:
-        text = re.sub(r"{([^\n{}]+?)}", r"{0[\1]}", template)
+        text = re.sub(r"{([^\n{}:]+?)}", r"{0[\1]}", template)
```
Comment (tmbo) [inline rasa/core/nlg/interpolator.py:27]: can you please add a test for this?

**Review by tmbo — DISMISSED** [review]
looks good, but should get a test before we merge

**Comment by tmbo** [comment]
@rgstephens do you have time to wrap this up?

File: rasa/core/nlg/interpolator.py (hunk 1 above)
Comment (rgstephens) [reply rasa/core/nlg/interpolator.py]: Added several tests which revealed issues in the original fix. Should be ready now.

File: tests/core/test_nlg.py [hunk 2]
```diff
@@ -296,6 +296,68 @@ def test_nlg_fill_template_button(button_slot_name, button_slot_value):
     }
 
 
+@pytest.mark.parametrize(
+    "button_slot_name, button_slot_value", [("button_1", "button_1_value")]
+)
+def test_nlg_fill_template_button_unquoted(button_slot_name, button_slot_value, caplog):
```
Comment (wochinge) [inline tests/core/test_nlg.py:302]: please add types to the parameters (same for the other tests)

File: tests/core/test_nlg.py [hunk 3]
```diff
@@ -296,6 +296,68 @@ def test_nlg_fill_template_button(button_slot_name, button_slot_value):
...
+        "buttons": [
+            {
+                "payload": f'/choose{{"some_slot":"{{{button_slot_name}}}"}}',
+                "title": f"{{{button_slot_name}}}",
+            }
+        ]
+    }
+    t = TemplatedNaturalLanguageGenerator(templates=dict())
+    caplog.set_level(logging.ERROR)
+    t._fill_template(
```
Comment (wochinge) [inline tests/core/test_nlg.py:315]: How about asserting the interplated template as well?

File: tests/core/test_nlg.py [hunk 4]
```diff
@@ -296,6 +296,68 @@ def test_nlg_fill_template_button(button_slot_name, button_slot_value):
     }
 
 
+@pytest.mark.parametrize(
```
Comment (wochinge) [inline tests/core/test_nlg.py:299]: what do you think of condensing all three tests to one using `pytest.mark.parametrize`?

File: changelog/7194.bugfix.md [hunk 5]
```diff
@@ -0,0 +1,2 @@
+Fix regex issue when processing payloads where intent and entity is
```
Comment (wochinge) [inline changelog/7194.bugfix.md:1]: Mhm, we actually don't change any regex in this PR, do we?

**Comment by wochinge** [comment]
Closing this for now. Please re-open if you continue this one.

**Pull request closed without merging**